protocol/0080-TBPM-time_based_pruning_mempool.md

```diff
@@ -0,0 +1,24 @@
+# Time based pruning of the mempool
+
+Transactions currently can stay in the mempool indefinitely. For non-validator commands, we want to be able to limit the time they are considered valid, once a certain amount of time has passed, the transaction should be rejected and removed from the mempool.
+
+## Network parameters
+
+- `network.transactions.maxttl` is a network parameter with type unsigned integer with the minimum value of `10` and a maximum value of `50`. This value determines the max number of blocks a given transaction is valid for. A transaction that specifies the `TTL`, this value *must* be less than, or equal to this parameter. If no `TTL` is specified, the transaction `TTL` will be set to this value.
```

Comment:
I understand that I have a maximum number of blocks that a transaction can be current for. 

How / where is the current block / block hash added to the transaction? Or is this not needed? Maybe the "current block and block hash" is already part of the transaction submitted and I am just not aware but it may be worth spelling out.

Reply:
Will clarify, but essentially, the transaction contains the block height at which the transaction was made. The TTL therefore will be `tx.InputData.BlockHeight + (tx.InputData.GoodForBlocks || netparams.MaxTTL)` where the TTL is either the specified number of blocks (for TxV4) or the max number of blocks defined by the network parameter

Reply:
> the transaction contains the block height 
Does it also contain the block hash? Otherwise the submitting client may lie about it if they want to mess with the chain.

Reply:
Yes, if the network is not specified when signing the transaction, there is _some_ leniency, but the tx hash + height both have to be provided, and should be fairly close to the current height/hash:

```go
	cmd.Flags().Uint64Var(&f.TxBlockHeight,
		"tx-height",
		0,
		"It should be close to the current block height when the transaction is applied, with a threshold of ~ - 150 blocks, not required if --network is set",
	)
	// ...
	cmd.Flags().StringVar(&f.TxBlockHash,
		"tx-block-hash",
		"",
		"The block-hash corresponding to tx-height which will be used to generate proof-of-work (hex encoded)",
	)


```

Reply:
Okay, I'll wait for you to clarify and then read again. Thx!

Reply:
Should be better now